### PR TITLE
Restructure text tokens in theme

### PR DESCRIPTION
**Pull request by origami-z**
Prep for type switch for theme next, where switch will depend on attribute on palette layer

- foundation tokens will be named
- new palette tokens with some semantic meaning can be token
- characteristics will be pointing to palette

### CODE REVIEW

File: .storybook/styles.css [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .salt-theme .sbdocs {
-  font-family: var(--salt-typography-fontFamily);
+  font-family: var(--salt-palette-text-fontFamily);
```
Comment (joshwooding) [inline .storybook/styles.css:2]: Should this be?

```suggestion
  font-family: var(--salt-text-fontFamily);
```

File: .storybook/styles.css [hunk 2]
```diff
@@ -41,7 +41,7 @@
   color: var(--salt-color-gray-900);
   font-size: var(--salt-text-fontSize);
   line-height: var(--salt-text-lineHeight);
-  font-family: var(--salt-typography-fontFamily);
+  font-family: var(--salt-palette-text-fontFamily-code);
```
Comment (joshwooding) [inline .storybook/styles.css:44]:
```suggestion
  font-family: var(--salt-text-code-fontFamily);
```
And would this be?

File: packages/lab/stories/skip-link/skip-link.stories.css [hunk 3]
```diff
@@ -4,6 +4,6 @@
 
 .goalsList {
   font-size: var(--salt-text-fontSize);
-  font-family: var(--salt-typography-fontFamily);
+  font-family: var(--salt-palette-text-fontFamily);
```
Comment (joshwooding) [inline packages/lab/stories/skip-link/skip-link.stories.css:7]:
```suggestion
  font-family: var(--salt-text-fontFamily);
```

?

File: tooling/css-variable-docgen-components/src/CharacteristicUsage.css [hunk 4]
```diff
@@ -1,5 +1,5 @@
 .characteristicTokenDoc {
-  font-family: var(--salt-typography-fontFamily-code);
+  font-family: var(--salt-palette-text-fontFamily-code);
```
Comment (joshwooding) [inline tooling/css-variable-docgen-components/src/CharacteristicUsage.css:2]:
```suggestion
  font-family: var(--salt-text-fontFamily-code);
```

**Review by joshwooding — APPROVED** [review]

**Review by joshwooding — DISMISSED** [review]

**Review by joshwooding — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Review by pseys — APPROVED** [review]

**Pull request merged**